packages/flutter_tools/lib/src/build_system/targets/web.dart

```diff
@@ -481,22 +531,53 @@ self.addEventListener("activate", function(event) {
 // The fetch handler redirects requests for RESOURCE files to the service
 // worker cache.
 self.addEventListener("fetch", (event) => {
+  if ('${serviceWorkerStrategy.toString()}' == 'ServiceWorkerStrategy.onlineOnly') {
+    return event.respondWith(fetch(event.request));
+  }
   var origin = self.location.origin;
   var key = event.request.url.substring(origin.length + 1);
   // Redirect URLs to the index.html
   if (event.request.url == origin || event.request.url.startsWith(origin + '/#')) {
     key = '/';
   }
-  // If the URL is not the RESOURCE list, skip the cache.
   if (!RESOURCES[key]) {
     return event.respondWith(fetch(event.request));
   }
-  event.respondWith(caches.open(CACHE_NAME)
+  if ('${serviceWorkerStrategy.toString()}' == 'ServiceWorkerStrategy.offlineFirst') {
+    return offlineFirst(event);
+  }
+  return onlineFirst(event);
+});
+
+// Attempt to download the resource online before falling back to
+// the offline cache.
+function onlineFirst(event) {
+  return event.respondWith(
+    fetch(event.request).then((response) => {
+      return caches.open(CACHE_NAME).then((cache) => {
+        cache.put(event.request, response.clone());
+        return response;
+      });
+    }).catch((error) => {
+      return caches.open(CACHE_NAME).then((cache) => {
+        return cache.match(event.request).then((response) => {
+          if (response != null) {
+            return response;
+          }
+          throw error;
```

Comment:
nit: I'd probably write this as:

```javascript
if (!response) {
  throw error;
}
return response;
```
 So that the "normal" flow is outside the branch.